flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -844,7 +844,7 @@ else if (current == CANCELING || current == RUNNING || current == DEPLOYING) {
 				// failing in the meantime may happen and is no problem.
 				// anything else is a serious problem !!!
 				if (current != FAILED) {
-					String message = String.format("Asynchronous race: Found state %s after successful cancel call.", state);
+					String message = String.format("Asynchronous race: Found %s in state %s after successful cancel call.", vertex.getTaskNameWithSubtaskIndex(), state);
 					LOG.error(message);
```

Comment:
nit: slf4j's `{}` placeholders should be used.

Reply:
Not really, because we reuse the message in the line below. Moreover, the logging statement is error and thus, will be evaluated in almost all cases. What one could argue is whether normal string concatenation wouldn't be faster than `String.format`.

Reply:
Ok